cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/DefaultStore.java

```diff
@@ -888,6 +891,38 @@ public List<ProgramHistory> getRuns(Collection<ProgramId> programs, ProgramRunSt
     });
   }
 
+  @Override
+  public Optional<byte[]> getState(AppState request) {
+    return TransactionRunners.run(transactionRunner, context -> {
+      return getAppStateTable(context).getState(request);
+    });
+  }
+
+  @Override
+  public void saveState(AppState request) {
+    TransactionRunners.run(transactionRunner, context -> {
+      getAppStateTable(context).saveState(request);
+    });
+  }
+
+  @Override
+  public void deleteState(AppState request) {
+    TransactionRunners.run(transactionRunner, context -> {
+      getAppStateTable(context).deleteState(request);
+    });
+  }
+
+  @Override
+  public void deleteAllStates(AppState request) {
```

Comment:
also need to modify the removeApp() and removeAll() methods to delete the state for the app